ql/src/test/results/clientpositive/perf/tez/constraints/cbo_query74.q.out

```diff
@@ -147,7 +147,7 @@ HiveSortLimit(sort0=[$2], sort1=[$0], sort2=[$1], dir0=[ASC], dir1=[ASC], dir2=[
                           HiveProject(ss_sold_date_sk=[$0], ss_customer_sk=[$3], ss_net_paid=[$20])
                             HiveTableScan(table=[[default, store_sales]], table:alias=[store_sales])
                       HiveProject(d_date_sk=[$0])
-                        HiveFilter(condition=[AND(=($1, 1999), IN($1, 1998, 1999))])
+                        HiveFilter(condition=[=($1, 1999)])
```

Comment:
I see that this patch works - but would have thinked that we didn't need something like this anymore - because IN is opened in an early phase - so Calcite should see a bunch of ANDs and ORs - and if that's true - RexSimplify could make this simplification - and there would be no need to enhance PointLookup...

I wonder how the IN was retained ...or re-created...

Reply:
What seems to happen is that we close these IN clauses before we call the join propagation rule, which leads to the inference of new predicates. Then the RexSimplify does not deal with the IN+EQUALS clauses and the HivePointLookup cannot deal with the EQUALS.
We could possibly change the rules order too but I was not brave enough to do that :) Plus I think supporting the degenerate case of EQUALS in HivePointLookup should be fine.
Iirc there is work going on in Calcite to finally have more extensive support for IN in RexSimplify, which you suggested some time ago... I hope that work goes in and we can simplify this whole code path.